model/jpa/src/main/resources/META-INF/jpa-changelog-24.0.0.xml

```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<!--
+  ~ * Copyright 2024 Red Hat, Inc. and/or its affiliates
+  ~ * and other contributors as indicated by the @author tags.
+  ~ *
+  ~ * Licensed under the Apache License, Version 2.0 (the "License");
+  ~ * you may not use this file except in compliance with the License.
+  ~ * You may obtain a copy of the License at
+  ~ *
+  ~ * http://www.apache.org/licenses/LICENSE-2.0
+  ~ *
+  ~ * Unless required by applicable law or agreed to in writing, software
+  ~ * distributed under the License is distributed on an "AS IS" BASIS,
+  ~ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ * See the License for the specific language governing permissions and
+  ~ * limitations under the License.
+  -->
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet author="keycloak" id="24.0.0-9758">
+        <addColumn tableName="USER_ATTRIBUTE">
+            <column name="LONG_VALUE" type="NCLOB" />
```

Comment:
Is it possible this to be different per database?
PostgreSQL has field text which is unlimited and with very good search performance.
We already use text field in database to overcome User attribute limit.

Unfortunately, I do not know if something similar exists in other databases.

Reply:
AFAIK liquibase already handles datatypes differently for each database. For PostgreSQL it seems it is indeed a text field: https://github.com/liquibase/liquibase/blob/043acd1581b695ad19c484ee985c6283666040d0/liquibase-standard/src/main/java/liquibase/datatype/core/ClobType.java#L104

Would it suffice for your use case?

Reply:
Yes. Nice.